services/motion/explore/explore.go

```diff
@@ -366,7 +357,7 @@ func (ms *explore) checkForObstacles(
 			// TODO: Generalize this fix to work for maps with non-transient obstacles. This current implementation
 			// relies on the plan being two steps: a start position and a goal position.
 			// JIRA Ticket: https://viam.atlassian.net/browse/RSDK-5964
-			plan[0][kb.Name().ShortName()] = currentInputs
+			plan.Trajectory()[0][kb.Name().ShortName()] = currentInputs
```

Comment:
I don't think this is safe for all possible implementations of `Trajectory()`.

 The *rrtPlan happens to return a slice of maps, rather than a copy of the slice of maps, which means that this happens to work for our implementation, however there is nothing about the documentation of the `Trajectory()` function that communicates that it should return a pointer which can be mutated at any time & which should then change the result of calling `Path()`
  
 I think this is a bad idea.

Reply:
I agree, but I dont have context for why this was happening hoping @jeremyrhyde can weigh in

Reply:
This is based on conversations with Peter and Nick, summarized here: https://github.com/viamrobotics/rdk/pull/3218/files#r1417689512

Reply:
My understanding of @nicksanford 's comment is not that we shouldn't be assigning `currentInputs` somewhere, but more specifically that the update from `plan` to `plan.Trajectory()` means that we now rely on the unknown implementation of `Trajectory()` returning something that is *persistently editable*, and we're relying on side-effects.

The solution would look like
```
planTraj := plan.Trajectory()
planTraj[0][kb.Name().ShortName()] = currentInputs
```
and then passing this new local object to CheckPlan

Reply:
changed to reflect this.  If everyones good with this I will resolve

Reply:
That works for me!